### DISCUSSION TITLE
I need to upload more than 10k records to a table

### DISCUSSION THREAD

**Discussion by jossmanbr2022** [pocketbase/pocketbase#824] [Q&A]
I need to upload more than 10k records to a table as would be the best option, I tried to do a forEach and it did not work
Giving more context I need to make a database with postal codes of Mexico and the USA to later make the query and filter the postal code

**Comment by hAhl2** [comment]
> I need to upload more than 10k records to a table as would be the best option, I tried to do a forEach and it did not work Giving more context I need to make a database with postal codes of Mexico and the USA to later make the query and filter the postal code

Here's an overview of what I did to upload a lot of records.

1)  Run Pocket Base, so that it creates the SQLite db.
2)  Download the SQLite executable
3)  Format the 10k records as a SQL insert script (.sql file), make sure to include an ID, createAt, updatedAt fields in the script
4)  using the SQLite command, execute the script against the SQLite db that pocketbase created
5)  check pocketbase to see if you inserterd the records correctly

**Comment by ganigeorgiev** [comment]
@jossmanbr2022 

>  I tried to do a forEach and it did not work

Why? Was there any error? I guess if you are not awaiting the records then autocancellation may got triggered. You could try the following options:

```js
// if the order is important, await each insert (this may also prevent SQLITE_BUSY/LOCKED errors)
async function importRecords(records) {
    for (let record of records) {
        await client.records.create("example", record);
    }
}

// OR make bulk insert requests
function bulkImportRecords(records) {
    const promises = [];
    
    for (let record of records) {
        const promise = client.records.create("example", record, {
            '$autoCancel': false,
        });

        promises.push(promise);
    }

    return Promise.all(promises);
}
```

---

Alternatively, you can prepare a sqlite dump file with your records as @hahl2 suggested.

**Reply by jossmanbr2022** [reply]
Thank you for your answers, I will try to correct the problem again, thanks
